src/agent/src/watcher.rs

```diff
@@ -80,16 +82,30 @@ impl Drop for Storage {
 }
 
 async fn copy(from: impl AsRef<Path>, to: impl AsRef<Path>) -> Result<()> {
-    if fs::symlink_metadata(&from).await?.file_type().is_symlink() {
+    let mut metadata = fs::symlink_metadata(&from).await?;
+    let mut gid = metadata.gid();
```

Comment:
Nit: to keep things symmetric, you can do:

```rust
let uid;

if condition {
    uid = 1;
} else {
    uid = 2;
}

```

Reply:
Thanks Maksym, I've actually further simplified this without needing the if else since calling `symlink_metadata` on a non-symlink file gets the same result as calling `metadata` so we don't need the if else check and can simply just use the `metadat.uid()` and `metadata.gid()`.